test/helpers/index.js

```diff
@@ -54,3 +55,9 @@ async function rmdir (dir) {
     throw error
   }
 }
+
+function localIP () {
+  const ifaces = Object.values(os.networkInterfaces()).flat()
+  const addrs = ifaces.filter(iface => !iface.internal && iface.family === 'IPv4')
+  return addrs.length ? addrs[0].address : null
+}
```

Comment:
would be nice to not have this, as we'd wanna run this in places with no os

Reply:
It's just for testing, I can change this:
https://github.com/holepunchto/serve-drive/blob/aa0630e902aab90813899e24dc2c9eb7b8f29669/test/all.js#L88

To this:
```js
if (!host) {
  t.comment('No local address')
  return
}
```

Or you say `localIP()` will somehow fail in those places?

Reply:
would be nice not to depend on `os` is what i am saying (subtext, is this worth testing)

Reply:
It's a current bug in main right now, but we know it works after this PR so we can remove the test

We use `os.tmpdir()` but we can create a tmp dir in cwd to completely remove `os`

Reply:
tmpdir is okay, easy to polyfill, but the network addresses is much more complicated and we rely on udx for that everywhere, so best to avoid that

Reply:
So we use UDX to get network interfaces, yes? Will do this in a different PR, plus the removal of the `URL` usage